ts/main.ts

```diff
@@ -79,6 +135,24 @@ function levelNumberToString(level: number): string {
   }
 }
 
+function switchCardsList(list: string): void {
+  if (list === 'all cards') {
+    if (!$spellsListFilteredCardsDiv.className.includes('hidden')) {
+      $spellsListFilteredCardsDiv.className += ' hidden';
+    }
+    if ($spellsListCardsDiv.className.includes('hidden')) {
+      $spellsListCardsDiv.classList.remove('hidden');
+    }
+  } else if (list === 'filtered cards') {
+    if (!$spellsListCardsDiv.className.includes('hidden')) {
+      $spellsListCardsDiv.className += ' hidden';
+    }
+    if ($spellsListFilteredCardsDiv.className.includes('hidden')) {
+      $spellsListFilteredCardsDiv.classList.remove('hidden');
+    }
+  }
+}
+
```

Comment:
Depending on the changes below will it be necessary to have different views for filtered cards and all of the cards? If we can make the changes then we would just need one view for all of the cards. This function would maybe still be needed as your view swap, but for spell details and spell books.